File: client/scalaz/src/main/scala/org/openapitools/client/api/QrsResponsegroups.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import QrsResponsegroups._

case class QrsResponsegroups (
  name: Option[String],
label: Option[String],
description: Option[String],
effectiveDate: Option[String],
untilDate: Option[String],
registrationStatus: Option[String],
version: Option[String],
qrsType: Option[String],
links: Option[QrsResponsegroupsLinks])

object QrsResponsegroups {
  import DateTimeCodecs._

  implicit val QrsResponsegroupsCodecJson: CodecJson[QrsResponsegroups] = CodecJson.derive[QrsResponsegroups]
  implicit val QrsResponsegroupsDecoder: EntityDecoder[QrsResponsegroups] = jsonOf[QrsResponsegroups]
  implicit val QrsResponsegroupsEncoder: EntityEncoder[QrsResponsegroups] = jsonEncoderOf[QrsResponsegroups]
}

File: client/clojure/src/cdisc_library_api/specs/qrs_responsegroups.clj
(ns cdisc-library-api.specs.qrs-responsegroups
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.qrs-responsegroups-links :refer :all]
            )
  (:import (java.io File)))


(def qrs-responsegroups-data
  {
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :description) string?
   (ds/opt :effectiveDate) string?
   (ds/opt :untilDate) string?
   (ds/opt :registrationStatus) string?
   (ds/opt :version) string?
   (ds/opt :qrsType) string?
   (ds/opt :_links) qrs-responsegroups-links-spec
   })

(def qrs-responsegroups-spec
  (ds/spec
    {:name ::qrs-responsegroups
     :spec qrs-responsegroups-data}))
